src/topology/bornology/basic.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import order.filter.cofinite
+
+/-!
+# Basic theory of bornology
+
+We develop the basic theory of bornologies. Instead of axiomatizing bounded sets and defining
+bornologies in terms of those, we recognize that the cobounded sets form a filter and define a
+bornology as a filter of cobounded sets which contains the cofinite filter.  This allows us to make
+use of the extensive library for filters, but we also provide the relevant connecting results for
+bounded sets.
+-/
+
+open set filter
+
+variables {α β : Type*}
+
+
+/-- A **bornology** on a type `α` is a filter of cobounded sets which contains the cofinite filter.
+Such spaces are equivalently specified by their bounded sets, see `bornology.of_bounded`
+and `bornology.ext_iff_is_bounded`-/
+@[ext]
+class bornology (α : Type*) :=
+(cobounded [] : filter α)
+(le_cofinite [] : cobounded ≤ cofinite)
+
+attribute [class] bornology
+
+/-- A constructor for bornologies by specifying the bounded sets,
+and showing that they satisfy the appropriate conditions. -/
+@[simps]
+def bornology.of_bounded {α : Type*} (B : set (set α))
+  (empty_mem : ∅ ∈ B) (subset_mem : ∀ s₁ ∈ B, ∀ s₂ : set α, s₂ ⊆ s₁ → s₂ ∈ B)
+  (union_mem : ∀ s₁ s₂ ∈ B, s₁ ∪ s₂ ∈ B) (sUnion_univ : ⋃₀ B = univ) :
+  bornology α :=
+{ cobounded :=
+  { sets := {s : set α | sᶜ ∈ B},
+    univ_sets := by { rwa ←compl_univ at empty_mem },
```

Comment:
You don't have to change this, but if you want you can remove these curly braces:
```suggestion
    univ_sets := by rwa ←compl_univ at empty_mem,
```